bambou/nurest_session.py

```diff
@@ -146,32 +152,21 @@ def _authenticate(self):
         self.login_controller.api_key = self._root_object.api_key
         bambou_logger.debug("[NURESTSession] Started session with username %s in enterprise %s" % (self.login_controller.user, self.login_controller.enterprise))
 
-    def _in_with_statement(self, frame):
-        """
-        Well, you'll have to trust me on this one.
-        """
-        return opname[ord(frame.f_code.co_code[frame.f_lasti + 3])] is "SETUP_WITH"
-
     def start(self):
         """
             Starts the session.
 
             Starting the session will actually get the API key of the current user
         """
 
-        try:
-            frame = inspect.stack()[1][0]
-        except IndexError:
-            _NURESTSessionCurrentContext.session = self
-            self._authenticate()
-            return self
+        if NURESTSession.session_stack:
+            bambou_logger.critical("VSPK currently does not support starting a session using start() inside a with statement. Please open all sessions outside the outermost with statement. Exiting for safety")
+            sys.exit(-1)
```

Comment:
Why not raising an exception here instead?

Reply:
agree